flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GroupAggProcessFunction.scala

```diff
@@ -129,17 +129,19 @@ class GroupAggProcessFunction(
       state.update(accumulators)
       cntState.update(inputCnt)
 
-      // if this was not the first row and we have to emit retractions
-      if (generateRetraction && !firstRow) {
+      // if this was not the first row and we may have to emit retractions
```

Comment:
I'd remove the `"and we may have to emit retractions"` part of the comment as this is not reflected by the condition.

Reply:
I've removed this comment.